### PR TITLE
Add hover interaction states to NxSegmentedButton & NxIconDropdown- RSC-1555

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://issues.sonatype.org/browse/RSC-1555

### CODE REVIEW

File: gallery/visualtests/nxBreadcrumb.js
Comment (EmilyBermudez88) [inline gallery/visualtests/nxBreadcrumb.js:39]: Added because the button was showing its hover state instead of selected state

**Comment by rpokorny** [comment]
Shouldn't this wait until dark mode is merged? The way you're using swatches directly in the `NxSegmentedButton.scss` files seems like the kind of think that isn't going to work once dark mode is merged

**Comment by EmilyBermudez88** [comment]
> Shouldn't this wait until dark mode is merged? The way you're using swatches directly in the `NxSegmentedButton.scss` files seems like the kind of think that isn't going to work once dark mode is merged

Yeah you're probably right that it would cause some problems and then another ticket to fix what's here. I'll convert it to a draft and hold off for now.

**Comment by rpokorny** [comment]
This part of the mockups (and similar on the others) is no longer correct, right?

image

**Comment by rpokorny** [comment]
> This part of the mockups (and similar on the others) is no longer correct, right?

Oh nevermind I see, that's the non-hover + focus state

File: gallery/visualtests/nxBreadcrumb.js
Comment (rpokorny) [reply gallery/visualtests/nxBreadcrumb.js:40]: > This has created a problem in the snapshots - if we use moveMouseAway(), then the tooltip begins to disappear when the snapshot is taken, or if we remove moveMouseAway() the color of the icon only button is in the hover state color. Is there a preference as to which we should use?

I'd remove it.  The color of the icon button isn't really the point of this screenshot; it's more about the custom width of the dropdown.

File: gallery/visualtests/nxSegmentedButton.js
Comment (rpokorny) [inline gallery/visualtests/nxSegmentedButton.js:75]: These tests seem to be intended to check when the toggle button is hovered and focused, but the screenshots show the toggle button hovered while a dropdown menu item is focused

Comment (EmilyBermudez88) [reply gallery/visualtests/nxSegmentedButton.js:75]: > These tests seem to be intended to check when the toggle button is hovered and focused, but the screenshots show the toggle button hovered while a dropdown menu item is focused

I realize I mixed up the two tests- hover + focus when only supposed to be hovered, and vice versa. When I wrote these tests during the interaction states epic I shifted focus to the dropdown menu to remove focus from the dropdown button- should I just blur the focus instead and remove any focus outlines?

Comment (EmilyBermudez88) [reply gallery/visualtests/nxSegmentedButton.js:73]: >  These tests seem to be intended to check when the toggle button is hovered and focused, but the screenshots show the toggle button hovered while a dropdown menu item is focused

As discussed in WW, the focus outlines were removed and snapshots were updated to reflect the proper interaction states. 
Segmented Button: [e5ad2f6](https://github.com/sonatype/sonatype-react-shared-components/pull/1120/commits/e5ad2f68e91ece2accdb3d4983ce696c6ee19e33)
Icon Dropdown: [5dac819](https://github.com/sonatype/sonatype-react-shared-components/pull/1120/commits/5dac819a83a215f5a709836db0671e501352dfdf)

File: gallery/visualtests/nxSegmentedButton.js
Comment (EmilyBermudez88) [reply gallery/visualtests/nxSegmentedButton.js:71]: > The "when hovered" screenshots appear to be showing focus styling as well

Fixed here: [e5ad2f6](https://github.com/sonatype/sonatype-react-shared-components/pull/1120/commits/e5ad2f68e91ece2accdb3d4983ce696c6ee19e33)

**Review by rpokorny — APPROVED** [review]

**Pull request merged**
